test/e2e/ic_conditions_metric_test.go

```diff
@@ -0,0 +1,136 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	routev1client "github.com/openshift/client-go/route/clientset/versioned"
+	"github.com/openshift/library-go/test/library/metrics"
+	prometheusv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/common/model"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+// waitForIngressControllerConditionsMetrics waits for the metrics for ingress_controller_conditions to be present.
+func waitForIngressControllerConditionsMetrics(t *testing.T, prometheusClient prometheusv1.API, start time.Time) error {
+	t.Logf("waiting for ingress_controller_conditions to be present")
+	if err := wait.PollUntilContextTimeout(context.TODO(), 1*time.Second, 2*time.Minute, false, func(context context.Context) (bool, error) {
+		r := prometheusv1.Range{
+			Start: start,
+			End:   time.Now(),
+			Step:  time.Minute,
+		}
+		result, _, err := prometheusClient.QueryRange(context, "ingress_controller_conditions", r)
+		if err != nil {
+			t.Logf("failed to fetch metrics: %v", err)
+			return false, nil
+		}
+
+		// Check if fetched metrics is of Vector type.
+		matrix, ok := result.(model.Matrix)
+		if !ok {
+			return ok, nil
+		}
+
+		// Check if length of returned metric Vector is zero.
+		if len(matrix) == 0 {
+			t.Logf("metric is empty... retrying")
+			return false, nil
+		}
+
+		return true, nil
+	}); err != nil {
+		return fmt.Errorf("error waiting for route metrics: %w", err)
+	}
+	return nil
+}
+func restartOperatorPod(t *testing.T, kubeClient kubernetes.Interface) time.Time {
+	// Find the operator pod
+	t.Logf("Restarting Ingress operator pod...")
+	var podsList *corev1.PodList
+	var err error
+	if podsList, err = kubeClient.CoreV1().Pods("openshift-ingress-operator").List(context.TODO(), metav1.ListOptions{}); err != nil || len(podsList.Items) == 0 {
+		t.Fatalf("failed to find the ingresscontroller operator pods: %v", err)
+	}
+	operatorPodName := podsList.Items[0].Name
+	if operatorPodName == "" {
+		t.Fatalf("unable to fetch Ingress Operator pod name")
+	}
```

Comment:
Just to make sure we are selecting the right pod:
```suggestion
	podsList, err := kubeClient.CoreV1().Pods("openshift-ingress-operator").List(context.Background(), metav1.ListOptions{})
	if err != nil {
		t.Fatalf("Failed listing pods in openshift-ingress-operator namespace: %v", err)
	}
	operatorPodName := ""
	for _, pod := range podsList.Items {
	    if strings.HasPrefix(pod.Name(), "ingress-operator") {
	        operatorPodName = pod.Name()
	        break
	    }
	}
	if operatorPodName == "" {
		t.Fatalf("Unable to find ingress operator pod")
	}
```